auxiliary/OpGasPriceOracle/contracts/OpGasPriceOracle.sol

```diff
@@ -73,20 +73,45 @@ contract OpGasPriceOracle is IExternalNode {
         RuntimeParams memory runtimeParams
     ) internal view returns (uint256 costOfExecutionGrossEth) {
         IOVM_GasPriceOracle ovmGasPriceOracle = IOVM_GasPriceOracle(ovmGasPriceOracleAddress);
+        bool isEcotone;
+        try ovmGasPriceOracle.isEcotone() returns (bool _isEcotone) {
+            isEcotone = _isEcotone;
+        } catch {
+            // If the call fails, we assume it's not an ecotone. Explicitly setting it to false to avoid missunderstandings
+            isEcotone = false;
+        }
+
+        if (isEcotone) {
+            // If it's an ecotone, use the new formula and interface
+            uint256 gasPriceL2 = ovmGasPriceOracle.baseFee();
+            uint256 baseFeeScalar = ovmGasPriceOracle.baseFeeScalar();
+            uint256 l1BaseFee = ovmGasPriceOracle.l1BaseFee();
+            uint256 blobBaseFeeScalar = ovmGasPriceOracle.blobBaseFeeScalar();
+            uint256 blobBaseFee = ovmGasPriceOracle.blobBaseFee();
+            uint256 decimals = ovmGasPriceOracle.decimals();
 
-        uint256 gasPriceL2 = ovmGasPriceOracle.baseFee();
-        uint256 baseFeeScalar = ovmGasPriceOracle.baseFeeScalar();
-        uint256 l1BaseFee = ovmGasPriceOracle.l1BaseFee();
-        uint256 blobBaseFeeScalar = ovmGasPriceOracle.blobBaseFeeScalar();
-        uint256 blobBaseFee = ovmGasPriceOracle.blobBaseFee();
-        uint256 decimals = ovmGasPriceOracle.decimals();
+            (uint256 gasUnitsL1, uint256 gasUnitsL2) = getGasUnits(runtimeParams);
 
-        (uint256 gasUnitsL1, uint256 gasUnitsL2) = getGasUnits(runtimeParams);
+            uint256 l1GasPrice = (baseFeeScalar *
+                l1BaseFee *
+                16 +
+                blobBaseFeeScalar *
+                blobBaseFee) / (16 * 10 ** decimals);
 
-        uint256 l1GasPrice = (baseFeeScalar * l1BaseFee * 16 + blobBaseFeeScalar * blobBaseFee) /
-            (16 * 10 ** decimals);
+            costOfExecutionGrossEth = ((gasUnitsL1 * l1GasPrice) + (gasUnitsL2 * gasPriceL2));
+        } else {
+            // If it's not an ecotone, use the legacy formula and interface
```

Comment:
for the sake of consistency, not important thought, using `ovmGasPriceOracle.baseFee()` instead of `ovmGasPriceOracle.gasPrice`

Reply:
We also need to add the try/catch on the validation step i think @leomassazza ?
https://github.com/Synthetixio/synthetix-v3/blob/main/auxiliary/OpGasPriceOracle/contracts/OpGasPriceOracle.sol#L135

Reply:
We are not doing the try/catch in the verification and just checking the common interface. The intention is to at least know the contract address looks like the precompiled one